### PR TITLE
Analytics

### LINKED ISSUE #37: add analytics tools

### PR DESCRIPTION
**Pull request by bion**
closes #37

### CODE REVIEW

File: app/views/layouts/application.html.rb [hunk 1]
```diff
@@ -19,6 +19,8 @@ def content
 
         meta(property: "og:url", content: "http://www.superdelegation.com")
         meta(property: "og:image", content: 'http://i.imgur.com/vM4Kr5M.jpg')
+
+        render 'shared/analytics'
```
Comment (dbalatero) [inline app/views/layouts/application.html.rb]: js goes in the footer or it blocks render!

Comment (dbalatero) [inline app/views/layouts/application.html.rb]: that goes for `application.js` too while you're at it

File: app/views/shared/_analytics.html.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      ga('send', 'pageview');
+    JS
+  end
+
+  def heap_analytics
+    if Rails.env.production?
+      javascript <<-JS
+        window.heap=window.heap||[],heap.load=function(e,t){window.heap.appid=e,window.heap.config=t=t||{};var r=t.forceSSL||"https:"===document.location.protocol,a=document.createElement("script");a.type="text/javascript",a.async=!0,a.src=(r?"https:":"http:")+"//cdn.heapanalytics.com/js/heap-"+e+".js";var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(a,n);for(var o=function(e){return function(){heap.push([e].concat(Array.prototype.slice.call(arguments,0)))}},p=["addEventProperties","addUserProperties","clearEventProperties","identify","removeEventProperty","setEventProperties","track","unsetEventProperty"],c=0;c<p.length;c++)heap[p[c]]=o(p[c])};
+
+        heap.load("3220453361");
```
Comment (dbalatero) [inline app/views/shared/_analytics.html.rb]: Prefer one set of JS and insert the ID. Commit the ID or `ENV` it, whichever is fine (it's a public value).

File: app/views/layouts/application.html.rb (hunk 1 above)
Comment (bion) [inline app/views/layouts/application.html.rb]: Heap specifies that it loads in the `<head>` tag: https://heapanalytics.com/docs/installation#web

File: app/views/layouts/application.html.rb [hunk 3]
```diff
@@ -23,7 +22,11 @@ def content
         render 'shared/analytics'
       end
 
-      body { yield }
+      body do
+        yield
+
+        text javascript_include_tag 'application'
```
Comment (dbalatero) [inline app/views/layouts/application.html.rb]: `text()` renders `<` as `&lt;` I'm pretty sure

Comment (bion) [inline app/views/layouts/application.html.rb]: the JS is rendering fine onto the page

**Comment by dbalatero** [comment]
:ship:

**Pull request merged**